.github/workflows/pr_verification.yml

```diff
@@ -176,3 +177,129 @@ jobs:
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: ghcr.io/candlepin/gha_build
+          tags: pr-${{ github.event.pull_request.number }}
+          containerfiles: ./.github/containers/Dockerfile
+
+      - name: Push to registry
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          image: ghcr.io/candlepin/gha_build
+          tags: pr-${{ github.event.pull_request.number }}
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+  spec_tests:
+    name: spec tests
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:38
+      options: --privileged
+    permissions:
+      contents: read
+      packages: write
+    needs: build_image
+    strategy:
+      fail-fast: false
+      matrix:
+        database: [postgres, mariadb]
+        mode: [standalone, hosted]
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        shell: bash
+        run: |
+          dnf -y --setopt install_weak_deps=False update
+          dnf --setopt install_weak_deps=False install -y gettext jss docker
+
+      - name: Install docker compose
+        shell: bash
+        run: |
+          DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
+          mkdir -p $DOCKER_CONFIG/cli-plugins
+          curl -SL https://github.com/docker/compose/releases/download/v2.18.1/docker-compose-linux-x86_64 -o $DOCKER_CONFIG/cli-plugins/docker-compose
+          sudo chmod +x $DOCKER_CONFIG/cli-plugins/docker-compose
+
+      - name: Log in to the registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - if: matrix.mode == 'standalone'
+        name: Add standalone env variable
+        working-directory: ./.github/containers
+        shell: bash
+        run: echo "CANDLEPIN_STANDALONE=true" > .env
+
+      - if: matrix.mode == 'hosted'
+        name: Add hosted env variables
+        working-directory: ./.github/containers
+        shell: bash
+        run: |
+          echo "CANDLEPIN_STANDALONE=false" > .env
+          echo "MODULE_CONFIG_HOSTED_CONFIGURATION_MODULE=org.candlepin.hostedtest.AdapterOverrideModule" >> .env
+
+      - if: matrix.database == 'postgres'
+        name: Create Candlepin and Postgres containers
+        shell: bash
+        run: |
+          echo "CANDLEPIN_IMAGE=ghcr.io/candlepin/gha_build:pr-${{ github.event.pull_request.number }}" >> ./.github/containers/.env
+          echo "SPEC_RUNNER_IMAGE=ghcr.io/candlepin/gha_spec_runner:latest" >> ./.github/containers/.env
+          echo "SOURCE_LOCATION=${{ github.workspace }}" >> ./.github/containers/.env
+          BRIDGE_NETWORK=$(docker network ls --filter=name=github_network_ --format="{{ .Name }}")
+          echo "NETWORK=$BRIDGE_NETWORK" >> ./.github/containers/.env
```

Comment:
When we specify to create a Fedora container from the Ubuntu runner VM, Github creates the container and automatically creates a network for the container. Unfortunately I don't see a way of specifying that network name using the container configurations. This seems like a downside of Github performing "magic" to make creating the Fedora container easy:
```
  spec_tests:
    name: spec tests
    runs-on: ubuntu-latest
    container:
      image: fedora:38
```

This is a problem for us as we are then creating a DB and Candlepin container that need to be in the same network as the Fedora container so that the spec tests can send requests to the Candlepin container. So, this grabs the network name using docker and the containers that are created using docker compose then join that network.